fhirpathpy/__init__.py

```diff
@@ -35,11 +35,31 @@ def apply_parsed_path(resource, parsedPath, context=None, model=None, options=No
             (options or {}).get("userInvocationTable", {})
         ),
     }
+    
+    # Add trace callback if provided in options
+    if options and "trace_callback" in options:
+        ctx["trace_callback"] = options["trace_callback"]
+    
     node = do_eval(ctx, dataRoot, parsedPath["children"][0])
 
     # Resolve any internal "ResourceNode" instances.  Continue to let FP_Type
     # subclasses through.
 
+    if options and options.get("return_raw_data", False):
+        if isinstance(node, list):
+            res = []
+            # Filter out intenal representation of primitive extensions
+            # even in this raw data mode (as they are not a part of the output)
```

Comment:
Question: why do we need to remove primitive extensions?

Reply:
In general I agree, that primitive extensions are just a JSON representation and should not be presented as fhirpath output. To this approach I have some questions, at least, why it's not recursive, but if it's a specific case that is needed only for fhirpath lab integration - it makes sense. Let's resolve it

Reply:
Ahh, I just read your comment about the filtration for this later, and now I see that visit() filters out them resursively.
I suppose, returnRawData option should be handled in visit() instead

Reply:
I played a bit with your example (contact.name.descendants() and returnRawData=True) and now I see the use-case.
Looks like, it's a special case that is possible using descendants() when primitive extension is returned on the root level as list item. And this behaviour make sense. So, I accept it and merge